### PR TITLE
Fix FDA error when no Im data and phaseQuad plot

### LINKED ISSUE #26602: FDA plot type FFT All causes crashes
### Expected behavior
When changing plot type on frequency domain analysis to `Frequency FFT All` the plot should be updated propperly. 
### Actual behavior
When changing plot type to FFT All when multiple runs are loaded FDA raises an error
### Steps to reproduce the behavior
1. open workbench
1. open Frequency domian analysis
1. load runs 62260-62261
1. go to the transform tab.
1. turn off `Imaginary Data` and then calculate FFT
1. go to home tab, and change plot type to `Frequency FFT All`
1. an error is raised.
### Platforms affected
tested on windows

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The FDA GUI had a couple of bugs with plotting, this fixes them.

**To test:**
test 1:
open workbench
open Frequency domian analysis
load runs 62260-62261
go to the transform tab.
turn off Imaginary Data and then calculate FFT
go to home tab, and change plot type to Frequency FFT All

test 2
Go to the phaseQuad tab and generate some data
Go to the transform tab and do some FFT's of phasequad
Go to home tab and try the various FFT plot options
The phaseQuad data should show up

Fixes #26602 . 

---
In release notes
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def plot_standard_workspaces(self):
         workspace_list = self.get_workspaces_to_plot(
             self.context.group_pair_context.selected, self._view.if_raw(),
             self._view.get_selected())
+            self._view.get_selected(), workspace_list)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py]:
```suggestion
```

File: scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py [hunk 2]
```diff
@@ -22,10 +22,15 @@ class FFT(object):
 
     def __init__(self, ws_freq_name, Re_run, Re, Im_run, Im, phasequad):
         self.Re_run = Re_run
-        self.Im_run = Im_run
+
         self.ws_freq_name = ws_freq_name
         self.Re = Re
-        self.Im = Im
+        if Im == "" and Im_run == "":
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py]: What happens if one of these is an empty string and not the other or are they always tied together?

File: scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py [hunk 3]
```diff
@@ -75,11 +80,14 @@ def get_frequency_workspace_names(self, run_list, group, pair, phasequad, freque
...
             for runs in run_list:
                 for run in runs:
-                    if (int(run) == int(fft.Re_run) or int(run) == int(fft.Im_run)) and \
-                       (fft.Re in group or fft.Re in pair or fft.Im in group or fft.Im in pair) and name not in names:
+                    # check Re part
+                    if int(run) == int(fft.Re_run) and (fft.Re in group or fft.Re in pair) and name not in names:
                         names.append(name)
+                    # check Im part
+                    if fft.Im and int(run) == int(fft.Im_run) and (fft.Im in group or fft.Im in pair) and name not in names:
+                         names.append(name)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py]:
```suggestion
                        names.append(name)
```

File: scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py (hunk 2 above)
Comment (AnthonyLim23) [reply scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py]: should be tied together:

Im run is set https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py#L214

Im group/pair is from 
https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py#L221
which calls
https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py#L52

so both should return empty strings

**Review by Harrietbrown — COMMENTED** [review]
This solves the bug pretty well.  but I've noticed that when plotting FFT All it plots an Re, Im, and mod line, but no Im line when plotting `Frequency Im`.
Is that how it should be behaving?

**Review by Matthew-Andrew — COMMENTED** [review]
This PR resolves the unhandled exception issue with the GUI and is an overall improvement. The issue described above exists on master already and after discussion we have decided to create an issue to investigate this at a later time.

**Review by Matthew-Andrew — APPROVED** [review]

**Comment by Harrietbrown** [comment]
I have made the problem I found into issue: #26616

**Comment by martyngigg** [comment]
There is apparently an existing issue to add tests to FDA. I would recommend that be tackled sooner rather than later if possible. The kind of logic changed here is very easy to get wrong.

**Comment by martyngigg** [comment]
The failing test is one of the know unreliable tests.

**Comment by AnthonyLim23** [comment]
its my job today, to write the tests. Would have done it last week, except I was doing this.

**Pull request merged**